gpt_index/readers/download.py

```diff
@@ -33,49 +76,60 @@ def download_loader(
     Returns:
         A Loader.
     """
-    dirpath = ".modules"
+    dirpath = "modules"
```

Comment:
Hmm, I feel like having a relative path here that can't be customized by the user is a bit weird to think about. Because modules will be downloaded to the relative path where the user is calling gpt_index. If their application has multiple entry points from different folders, `import modules` will break.

Here's an alternative idea: what if we created a folder under gpt_index/readers called `llamahub_modules` or something, that comes with a pre-existing __init__.py file. Every module downloaded from llamahub will be downloaded to that directory. You can get the path to `llamahub_modules` from the Path to the current file as follows:

```
modules_dir = Path(__file__).parent / "llamahub_modules"
```

Then the user can import the modules as follows: `from gpt_index.readers.llamahub_modules import GithubRepositoryReader, ...`

Reply:
Good catch, I didn't actually think about this. I was going to suggest that we rename it from `modules` to `loaders` but I think `llamahub_modules` under `readers` is much better. 

So, if I understand it correctly, we are going to have existing `llamahub_modules` with existing `__init__.py` which we might have to modify in order to directly import from `llamahub_modules` instead of `llamahub_modules.<loader_name>`

in `download.py`, instead of hardcoded `modules` path we are going to construct the module based on the location of `download.py`. This is necessary because the `modules` would be at the same level as `gpt_index` which is entry module to `gpt_index`, so it might complicate the imports between user's modules?

Reply:
> So, if I understand it correctly, we are going to have existing llamahub_modules with existing __init__.py which we might have to modify in order to directly import from llamahub_modules instead of llamahub_modules.<loader_name>

If it's easier we could also just not modify the __init__.py and have users do the import directly e.g. `from gpt_index.readers.llamahub_modules.github_repository_reader import GithubRepositoryReader`

> in download.py, instead of hardcoded modules path we are going to construct the module based on the location of download.py. This is necessary because the modules would be at the same level as gpt_index which is entry module to gpt_index, so it might complicate the imports between user's modules?

Yeah in your approach, the parent directory to `modules` could actually be anywhere where the user is calling the gpt index code, not necessarily in the `gpt_index` folder. the user may not want the folder to be in that location

Reply:
@ahmetkca just a heads up, i'm happy to help take a stab at this

Reply:
@jerryjliu thanks, by the way sorry I could spare time to this during the weekday. I'll try to finalize this weekend. I have moved to a folder under `readers` with the name `llamahub_modules`. I haven't pushed it yet. I was mainly trying to import downloaded loaders directly from `from gpt_index.readers.llamahub_modules import x, y, z` by programmatically writing to `__init__.py` of `llamahub_modules` python module. So that users don't have to add it to `__init__.py` of `llamahub_modules` file. I just pushed a commit now. You can try it. The problem is that I can run `download_loader` from python console and I can import the downloaded loaders directly from `llamahub_modules`.
But the first time calling `download_loader` from a file and importing from like `from gpt_index.readers.llamahub_modules import x, y, z` is ends up with the following error:
```
Traceback (most recent call last):
  File "/home/ahmetk/Projects/download_loader-gpt_index/delme.py", line 17, in <module>
    from gpt_index.readers.llamahub_modules import GithubRepositoryReader
ImportError: cannot import name 'GithubRepositoryReader' from 'gpt_index.readers.llamahub_modules' (/home/ahmetk/Projects/download_loader-gpt_index/gpt_index/readers/llamahub_modules/__init__.py)
```

Reply:
Hey @ahmetkca, thanks for sharing the findings, I'll take a look! 

No pressure at all in terms of your availability/timeline - I will take a look at this later today, test it out, and see if I can push some changes